lib/explorer/shared.ex

```diff
@@ -27,7 +27,7 @@ defmodule Explorer.Shared do
   within lists inside.
   """
   def dtypes do
-    @non_list_types ++ [{:list, :any}]
+    @scalar_types ++ [{:list, :any}, {:struct, :any}]
```

Comment:
I followed a similar approach from the list type implementation and used `:any` to denote "a struct of any shape". I'm not 100% happy with it though, please let me know if you have any suggestions.